File: cpp/1.two-sum.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    unordered_map<int, int> hash;
    for (int idx = 0; idx < nums.size(); idx++) {
      auto it = hash.find(nums[idx]);
      if (it != hash.end()) {
        return {idx, hash[nums[idx]]};
      } else {
        hash[target - nums[idx]] = idx;
      }
    }
    // not found, return undefined
    return {0, 0};
  }
};

File: py/1.two-sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        for i, num in enumerate(nums):
            if target-num in hash:
                return [i, hash[target-num]]
            else:
                hash[num] = i
        # no answer
        return [0, 0]
